designs/2022-community-eslint-org/README.md

```diff
@@ -0,0 +1,328 @@
+same name on `npm`.  
+Even when we would release a package under the `@eslint-community` npm scope
+(for instance because we can't get the rights to publish to the existing npm
+package), people could still use the old package without any extra problems
+(except for the ones they already had) if they wanted to.
+
+## Alternatives
+
+<!--
+    What other designs did you consider? Why did you decide against those?
+
+    This section should also include prior art, such as whether similar
+    projects have already implemented a similar feature.
+-->
+
+- Private persons could create a fork of the above mentioned packages.  
+  This could however create ceveral well maintained forks, which would cause
+  fragmentation in the community.
+
+  More importantly, this could cause the "problem" of the fork being reliant on
+  that specific individual, which could put us in the same position
+  (abandoned/unmaintained package) all over again at some point in time.
+  Possibly having a new fork each X amount of time, would also have the burden
+  of needing to update all existing dependants each time.
+
+- The community could independently create multiple more focused organizations
+  to maintain a subset of ESLint related packages.  
+  Some examples of already existing organizations like that are
+  [@import-js](https://github.com/import-js) (which maintains
+  [`eslint-plugin-import`](https://github.com/import-js/eslint-plugin-import)),
+  [@jsx-eslint](https://github.com/jsx-eslint) (which maintains
+  [`eslint-plugin-jsx-a11y`](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y),
+  [`eslint-plugin-react`](https://github.com/jsx-eslint/eslint-plugin-react) &
+  [`jsx-ast-utils`](https://github.com/jsx-eslint/jsx-ast-utils)) &
+  [@standard](https://github.com/standard) (which maintains a couple of
+  [`standard`](https://github.com/standard/standard) related ESLint configs)
+
+## Open Questions
+
+<!--
+    This section is optional, but is suggested for a first draft.
+
+    What parts of this proposal are you unclear about? What do you
+    need to know before you can finalize this RFC?
+
+    List the questions that you'd like reviewers to focus on. When
+    you've received the answers and updated the design to reflect them,
+    you can remove this section.
+-->
+
+- What exact acceptance criteria do we want to use?
+- What's the main place of communication for the "core" team?
+- Will there be a single team collectively maintaining all modules within the
+  project? Or will each module possibly have their own maintainers?
+- What maintenance principles will be enforced?  
+  Will each module have their own standards, release flows, dependency update
+  checks etc or will it be streamlined across all modules?
+- How does this relate to stuff in eg. `@standard`?  
+  Would there be an expectency from the wider community or the
+  `@eslint-community` org itself to transfer some stuff from these places?
```

Comment:
👎 to accept shared configs - they can be very opinionated, hard to say which is better. it 
should only accept eslint-related infrastructures: parsers, plugins, formatters..., so the shared configs like standard can also benifit from it.

Reply:
Sometimes its not as easy to separate the two.

Standard eg. had a now deprecated [`eslint-plugin-standard`](https://github.com/standard/eslint-plugin-standard) (deprecated in favor of `eslint-plugin-node`: https://github.com/mysticatea/eslint-plugin-node/issues/178)

And it still has +1 million weekly downloads, still +1k dependents on npm and [676k repositories](https://github.com/standard/eslint-plugin-standard/network/dependents) depending on it on GitHub, so from metrics a package like that would actually qualify.

<img width="1197" alt="Skärmavbild 2022-06-02 kl  11 18 34" src="https://user-images.githubusercontent.com/34457/171598559-2a58d220-9d5e-426f-8cef-a9162f6385d6.png">

Also:

Most plugins have a shared recommended config in them.

The difference between these are a bit blurry:

```json
"extends": ["plugin:node/recommended"]
```

```json
"extends": ["standard"]
```

Reply:
I agree with not including shared configs. Anyone can create a duplicate shared config at any time via “extends”, so I don’t see a need to take over shareable configs.

Reply:
Extending them can still cause problems if these configs have (peer)dependencies that aren't compatible with the latest ESLint version or if they're referencing removed rules for instance.

Creating a fork on a personal account could solve this, but then we're again at the point that it's dependent on that one person again.

I agree that we should be more thought through when adding configs though.